includes/sanitizers/class-amp-core-theme-sanitizer.php

```diff
@@ -2069,43 +2069,34 @@ static function() {
 
 				$styles = file_get_contents( $css_file ); //phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
 
-				// Make the rules for the mobile menu less specific so that it can also be applied to the
-				// amp-lightbox mobile menu also.
-				$new_styles = str_replace( '.primary-navigation > .primary-menu-container', '.primary-navigation .primary-menu-container', $styles );
-				$new_styles = str_replace( '.primary-navigation > div > .menu-wrapper', '.primary-navigation div > .menu-wrapper', $new_styles );
-
 				// Append any extra rules that may be needed.
-				$new_styles .= '
-					.primary-navigation-open .menu-button-container {
-						/* Needs to be shown above amp-lightbox, which has a z-index of 1000. */
-						z-index: 1001;
-					}
-
+				$styles .= '
+					/* Trap keyboard navigation within mobile menu when it\'s open */
 					@media only screen and (max-width: 481px) {
-						/* Hide the desktop menu so that it does not interfere when the amo-lightbox mobile menu is open. */
-						.primary-navigation > .primary-menu-container {
-							display: none;
+						.primary-navigation-open #page {
+							visibility: hidden !important;
 						}
 
-						/* Accommodate for the admin bar height */
-						.admin-bar amp-lightbox .primary-menu-container {
-							margin-top: var(--global--admin-bar--height);
+						#primary-mobile-menu[aria-expanded="true"] {
+							visibility: visible;
```

Comment:
This should keep the focus within the mobile menu when it's open, unless there is something outside of `#page`. I attempted to override the inline `visibility: visible` style rule on the body element that AMP adds but that does not seem to be possible.

Reply:
Yeah, what you've done seems to work perfectly!

Is there a specific reason to use `!important` for `visibility: hidden`?

Reply:
That's left over after some debugging, will remove.